ftw/keywordwidget/testing.py

```diff
@@ -1,14 +1,29 @@
 from ftw.builder.testing import BUILDER_LAYER
 from ftw.builder.testing import functional_session_factory
 from ftw.builder.testing import set_builder_session_factory
+from ftw.keywordwidget.behavior import IKeywordUseCases
 from plone.app.testing import applyProfile
 from plone.app.testing import FunctionalTesting
 from plone.app.testing import PLONE_FIXTURE
 from plone.app.testing import PloneSandboxLayer
+from plone.indexer import indexer
 from plone.testing import z2
+from Products.CMFCore.interfaces import IContentish
+from zope.component import provideAdapter
 from zope.configuration import xmlconfig
 
 
+def _setup_catalog_for_tests(portal):
+    @indexer(IContentish)
+    def unicode_keywords(obj):
+        return IKeywordUseCases(obj).unicode_keywords
+
+    provideAdapter(unicode_keywords)
+
+    catalog = portal.portal_catalog
+    catalog.addIndex('unicode_keywords', 'KeywordIndex')
+
+
 class FtwLayer(PloneSandboxLayer):
     defaultBases = (PLONE_FIXTURE, BUILDER_LAYER)
```

Comment:
`_setup_catalog_for_tests` has a side effect: the provided adapter is not removed.
Although this is not problematic in this case, since it is actually used in all tests, it may cause trouble in the future.

We could simply use [`COMPONENT_REGISTRY_ISOLATION `](https://github.com/4teamwork/ftw.testing#component-registry-isolation-layer) instead of `PLONE_FIXTURE` so that proper isolation is ensured.